src/providers/storage/assetProviderFactory.ts

```diff
@@ -36,6 +38,14 @@ export class AssetProviderFactory {
             };
         }
 
+        if (options.platformSupport == null) {
```

Comment:
This would miss an undefined `platformSupport`.  Change to `if (!options.platformSupport)` which will catch both null and undefined values.